utils/encryption/encryption.go

```diff
@@ -0,0 +1,104 @@
+package encryption
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+
+	"golang.org/x/crypto/nacl/box"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+)
+
+const KeySize = 32
+
+var (
+	keySizeError = fmt.Errorf("%w: recipient key has invalid size (expected %d bytes)", commonerrors.ErrInvalid, KeySize)
+)
+
+// GenerateKeyPair generates a asymmetric key pair suitable for use with encryption utilities. Works with [NaCl box](https://nacl.cr.yp.to/box.html.)
+func GenerateKeyPair() (base64EncodedPublicKey, base64EncodedPrivateKey string, err error) {
```

Comment:
Maybe these should be a more complex type than a string. Maybe make it harder to read the keys by having them in a mutex or something and make getters compulsory so you can't just accidentally log the string containing the key.

I am not sure

Reply:
made it a more complex type